mllib/src/main/scala/org/apache/spark/ml/feature/VectorSlicer.scala

```diff
@@ -100,10 +100,10 @@ final class VectorSlicer @Since("1.5.0") (@Since("1.5.0") override val uid: Stri
     // Validity checks
     transformSchema(dataset.schema)
     val inputAttr = AttributeGroup.fromStructField(dataset.schema($(inputCol)))
-    inputAttr.numAttributes.foreach { numFeatures =>
+    if ($(indices).nonEmpty && inputAttr.size >= 0) {
```

Comment:
Do you want to check .size once? not that it probably matters as it's probably not a O(n) operation on this collection. Looks fine